client/src/components/WebAuthn/Register.js

```diff
@@ -36,17 +29,38 @@ class Register extends Component {
     this.handleStartRegistration = this.handleStartRegistration.bind(this);
   }
 
+/**
+ * We rely on a prop called `keyData` being present, however it may not be set upon the first
+ * render of the component. We can check and wait for it via state.
+ *
+ * React 16 deprecates `componentWillReceiveProps` replacing it with `getDerivedStateFromProps`
+ * for setting state in a functional manner. We have chosen to use {@see componentDidUpdate} over
+ * `componentWillReceiveProps` in this component in order to provide the same functionality across
+ * different (supported) build versions.
+ * By having one delegate to the other we still save in maintenance costs.
+ *
+ * @param {Object} newProps incoming props for this component
+ * @param {Object} currentState current stated of this component
+ */
+  static getDerivedStateFromProps(newProps, currentState) {
+    if (currentState.view === VIEWS.LOADING && newProps.keyData) {
+      return { view: VIEWS.READY };
+    }
+    return null;
+  }
+
   /**
-   * keyData may not be passed through on initial render, and we can't start without it.
-   *
-   * @todo Re-evaluate whether this code is required.
+   * React docs (as at React 16) state that one should avoid setting state in this method...
```

Comment:
>React docs (as at React 16) state that one should avoid setting state in this method

I can't see this in the docs: https://reactjs.org/docs/react-component.html#componentdidupdate

---

Do you need to use both getDerivedStateFromProps and componentDidUpdate? My suggestion was to use the latter only, removing the former and the cross-compatible code you'd added for it.

Reply:
That's implied context I've read into perhaps.

>You may call setState() immediately in componentDidUpdate() but note that it must be wrapped in a condition like in the example above, or you’ll cause an infinite loop.

![image](https://user-images.githubusercontent.com/778003/59802959-3bad9d80-933e-11e9-9b97-246f4d3d2e79.png)

Reply:
Would you like me to update this?

Reply:
Yes please, it's incorrect

Reply:
I updated it from "state one should avoid" to "imply one should try to avoid"

Reply:
Can you please provide a reference for this statement? Nothing I read in those docs said this at all

Reply:
I added the paragraph in the screenshot above.